service/service.go

```diff
@@ -0,0 +1,78 @@
+package service
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/giantswarm/microendpoint/service/version"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/viper"
+
+	"github.com/giantswarm/shutdown-deferrer/flag"
+)
+
+// Config represents the configuration used to create a new service.
+type Config struct {
+	Logger micrologger.Logger
+
+	Flag  *flag.Flag
+	Viper *viper.Viper
+
+	Description string
+	GitCommit   string
+	ProjectName string
+	Source      string
+}
+
+// Service is a type providing implementation of microkit service interface.
+type Service struct {
+	Version *version.Service
+
+	bootOnce sync.Once
+}
+
+// New creates a new service with given configuration.
+func New(config Config) (*Service, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+	config.Logger.Log("level", "debug", "message", fmt.Sprintf("creating shutdown-referrer gitCommit:%s", config.GitCommit))
```

Comment:
When we want to have something like this we should maybe build it into the `version` or `daemon` command somehow.

Reply:
Maybe. This just came with copy&paste from `release-operator`. I'll remove this.